### ISSUE TITLE
azure_rm_virualmashine issue

### ISSUE THREAD

**Issue by snemir2** [ansible/ansible-modules-core#4324]
##### ISSUE TYPE

- Bug Report
- Feature Idea
- Documentation Report
##### COMPONENT NAME

azure_rm_virtualmachine module
##### ANSIBLE VERSION

```
ansible-2.1.0.0-1.fc23.noarch

```
##### CONFIGURATION

Python 2.7.11
Modules:
azure (2.0.0rc5)
##### OS / ENVIRONMENT

fedora 23
##### STEPS TO REPRODUCE

```

---
 - hosts: localhost
   connection: local
   gather_facts: false
   become: false
   vars_files:
  # - environments/Azure/azure_credentials_encrypted.yml
    - ../../inventory/environments/Azure/azure_credentials_encrypted_temp_passwd.yml
   vars:

   roles:
    - create_azure_vm


And roles/create_azure_vm/main.yml 
- name: Create VM with defaults
  azure_rm_virtualmachine:
    resource_group: Testing
    name: testvm10
    admin_username: test_user
    admin_password: test_vm
    image:
          offer: CentOS
          publisher: OpenLogic
          sku: '7.1'
          version: latest

```

##### EXPECTED RESULTS

creatiion of VM. 
##### ACTUAL RESULTS

PLAYBOOK: provision_azure_playbook.yml *****************************************
1 plays in provision_azure_playbook.yml

PLAY [localhost] ***************************************************************

TASK [create_azure_vm : Create VM with defaults] *******************************
task path: /ansible/ansible_home/roles/create_azure_vm/tasks/main.yml:3
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: snemirovsky
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "`echo $HOME/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045`" && echo ansible-tmp-1470326423.51-208881287834045="`echo $HOME/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045`" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpiYFkuQ TO /home/snemirovsky/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045/azure_rm_virtualmachine
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/snemirovsky/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045/azure_rm_virtualmachine; rm -rf "/home/snemirovsky/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045/" > /dev/null 2>&1 && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py", line 1284, in <module>
    main()
  File "/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py", line 1281, in main
    AzureRMVirtualMachine()
  File "/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py", line 487, in **init**
    for key in VirtualMachineSizeTypes:
NameError: global name 'VirtualMachineSizeTypes' is not defined

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "azure_rm_virtualmachine"}, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py\", line 1284, in <module>\n    main()\n  File \"/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py\", line 1281, in main\n    AzureRMVirtualMachine()\n  File \"/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py\", line 487, in **init**\n    for key in VirtualMachineSizeTypes:\nNameError: global name 'VirtualMachineSizeTypes' is not defined\n", "module_stdout": "", "msg": "MODULE FAILURE", "parsed": false}

NO MORE HOSTS LEFT *************************************************************
        to retry, use: --limit @provision_azure_playbook.retry

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1

```
PLAYBOOK: provision_azure_playbook.yml *****************************************
1 plays in provision_azure_playbook.yml

PLAY [localhost] ***************************************************************

TASK [create_azure_vm : Create VM with defaults] *******************************
task path: /ansible/ansible_home/roles/create_azure_vm/tasks/main.yml:3
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: snemirovsky
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045 `" && echo ansible-tmp-1470326423.51-208881287834045="` echo $HOME/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpiYFkuQ TO /home/snemirovsky/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045/azure_rm_virtualmachine
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/snemirovsky/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045/azure_rm_virtualmachine; rm -rf "/home/snemirovsky/.ansible/tmp/ansible-tmp-1470326423.51-208881287834045/" > /dev/null 2>&1 && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py", line 1284, in <module>
    main()
  File "/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py", line 1281, in main
    AzureRMVirtualMachine()
  File "/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py", line 487, in __init__
    for key in VirtualMachineSizeTypes:
NameError: global name 'VirtualMachineSizeTypes' is not defined

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "azure_rm_virtualmachine"}, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py\", line 1284, in <module>\n    main()\n  File \"/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py\", line 1281, in main\n    AzureRMVirtualMachine()\n  File \"/tmp/ansible_Xf2rTh/ansible_module_azure_rm_virtualmachine.py\", line 487, in __init__\n    for key in VirtualMachineSizeTypes:\nNameError: global name 'VirtualMachineSizeTypes' is not defined\n", "module_stdout": "", "msg": "MODULE FAILURE", "parsed": false}

NO MORE HOSTS LEFT *************************************************************
        to retry, use: --limit @provision_azure_playbook.retry

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1

```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by snemir2** [comment]
tried with azure module rc2 and rc4 with same results.

**Comment by chapmanb** [comment]
I ran into this as well and it's a bit difficult to debug since the entire import section of the module is enclosed in a try/except:

https://github.com/ansible/ansible-modules-core/blob/e5c530a4884aff9783181474c9112222b38e2ee0/cloud/azure/azure_rm_virtualmachine.py#L442

If any imports fail the module won't end up loading `VirtualMachineSizeTypes`, since it's at the end of the block, leading to this error. Practically, I changed my local version to re-raise the error instead of swallowing it and that pointed out the missing module from my install (`msrestazure` in my case). Hopefully that debugging approach helps resolve your issue.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ghost** [comment]
Same error here

**Comment by nitzmahone** [comment]
@ulrith have you tried upgrading the msrestazure package as suggested above? That's been the source of a lot of issues against RC5 (`pip install msrestazure --upgrade`)

**Comment by ghost** [comment]
@nitzmahone Yes it works, thank you!

**Comment by nitzmahone** [comment]
Closing: upstream issue solved by Microsoft SDK update; no need for docs once Azure SDK hits release (next few weeks, last I heard).

**Comment by rajeshbshinde** [comment]
@nitzmahone I am using Azure Ansible version 2.0.0.rc5 and also tried to run above msrestazure upgrade command. I am still facing this issue. NameError: global name 'VirtualMachineSizeTypes' is not defined

**Comment by elaurijssens** [comment]
With 2.0.0rc5 msrestazure is not installed as a dependency. The error goes away if you install it, but then it doesn't support machine types like Standard_A4_v2.

**Comment by dagwieers** [comment]
@nitzmahone Does this issue require a note in the module explaining the requirements (azure == 2.0.0rc5) better ? Or is there a new similar issue at work ?

I am going to reopen this issue so this can be re-evaluated.

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by dagwieers** [comment]
Uh-oh, this is the old repository.. @elaurijssens @rajeshbshinde If this is still an issue, please open a new ticket at https://github.com/ansible/ansible/issues, or migrate this ticket using: https://github-issue-mover.appspot.com/

**Comment by elaurijssens** [comment]
@dagwieers i noticed, will do.
